examples/src/demo4_obj_fun.cxx

```diff
@@ -0,0 +1,153 @@
+//
+//
+/*!
+  \file
+  \ingroup examples
+  \brief An example of a method to compute the objective function value
+  of an image and then perform basic iterative gradient ascent.
+  All parameters are parsed from a parameter file.
+
+  It illustrates
+	- basic class derivation principles
+	- how to use ParsingObject to have automatic capabilities of parsing
+	  parameters files (and interactive questions to the user)
+	- how to initialise and setup a objective function object
+    - how to compute the objective function (log-likelihood) gradient
+    - how a basic iterative optimisation works (or may not, if parameters are altered)
+
+  Note that the same functionality could be provided without deriving
+  a new class from ParsingObject. One could have a KeyParser object
+  in main() and fill it in directly.
+
+  See README.txt in the directory where this file is located.
+
+  \author Kris Thielemans and Robert Twyman      
+*/
+/*
+    Copyright (C) 2020 University College London
+
+    This software is distributed under the terms 
+    of the GNU General  Public Licence (GPL)
+    See STIR/LICENSE.txt for details
+*/
+
+#include "stir/IO/OutputFileFormat.h"
+#include "stir/recon_buildblock/PoissonLogLikelihoodWithLinearModelForMeanAndProjData.h"
+#include "stir/IO/read_from_file.h"
+#include "stir/is_null_ptr.h"
+
+namespace stir {
+
+class MyStuff: public ParsingObject
+{
+public:
+  void set_defaults();
+  void initialise_keymap();
+  bool post_processing();
```

Comment:
make all these private. Add a default constructor that calls `set_defaults()`

Reply:
Doing the above, calling `set_defaults()` from constructor, when `set_defaults()` is private/protected wont work. It needs to be public.